### PR TITLE
Make LUISGen more robust to the presence/absence of features.

### LINKED ISSUE #1107: LuisGen  RuntimeBinderException: Cannot perform runtime binding on a null reference
## Tool
Name: LuisGen
Version: 2.0.2
OS: Win 10x64
## Describe the bug

RuntimeBinderException on json file

## To Reproduce
1. "luis export version --appId xxx --versionId "0.1" --authoringKey xxxx > general.json" this i have run on the general Luis app created using enterprise template bot 
2.  luisgen general.json -cs general 

Which is giving me an error of 

Generating file general.cs that contains class Luis.general.

Unhandled Exception: Microsoft.CSharp.RuntimeBinder.RuntimeBinderException: Cannot perform runtime binding on a null reference
   at CallSite.Target(Closure , CallSite , Object )
   at System.Dynamic.UpdateDelegates.UpdateAndExecute1[T0,TRet](CallSite site, T0 arg0)
   at CallSite.Target(Closure , CallSite , Object )
   at LUISGen.Utils.WriteInstances(JObject obj, Action`1 writeInstance) in D:\a\1\s\packages\LUISGen\src\Utils.cs:line 108
   at System.Dynamic.UpdateDelegates.UpdateAndExecuteVoid3[T0,T1,T2](CallSite site, T0 arg0, T1 arg1, T2 arg2)
   at LUISGen.Utils.WriteInstances(JObject obj, Action`1 writeInstance) in D:\a\1\s\packages\LUISGen\src\Utils.cs:line 125
   at LUISGen.CSharp.Entities(Object app, Writer w) in D:\a\1\s\packages\LUISGen\src\CSharp.cs:line 212
   at System.Dynamic.UpdateDelegates.UpdateAndExecuteVoid3[T0,T1,T2](CallSite site, T0 arg0, T1 arg1, T2 arg2)
   at LUISGen.CSharp.Generate(String description, Object app, String className, String space, String outPath) in D:\a\1\s\packages\LUISGen\src\CSharp.cs:line 272
   at System.Dynamic.UpdateDelegates.UpdateAndExecuteVoid6[T0,T1,T2,T3,T4,T5](CallSite site, T0 arg0, T1 arg1, T2 arg2, T3 arg3, T4 arg4, T5 arg5)
   at LUISGen.Program.Main(String[] args) in D:\a\1\s\packages\LUISGen\src\Program.cs:line 151

### PR DESCRIPTION
**Pull request by chrimc62**
Fixes #1107 

## Proposed Changes
Made LUISGen robust to properties missing from export file.

### CODE REVIEW

File: packages/LUISGen/src/Utils.cs [hunk 1]
```diff
@@ -104,26 +104,29 @@ private static void WriteInstances(dynamic entities, Action<string> writeInstanc
 
         public static void WriteInstances(JObject obj, Action<string> writeInstance)
         {
-            dynamic app = obj;
-            if (app.entities != null)
+            if (obj != null)
```
Comment (cleemullins) [inline packages/LUISGen/src/Utils.cs]: Unit tests for this function are needed.

**Comment by cleemullins** [comment]
Please consider adding unit tests to the code that's changed.

File: packages/LUISGen/src/Utils.cs [hunk 2]
```diff
@@ -35,16 +35,16 @@ public static bool IsPrebuilt(dynamic name, dynamic app)
         public static bool IsList(dynamic name, dynamic app)
         {
             bool isList = false;
-            if (app.closedLists != null)
+            if (app?.closedLists != null)
```
[the remark this answers is not in the document]
Comment (chrimc62) [reply packages/LUISGen/src/Utils.cs]: This generates code.  How would you recommend doing a test?  I regenerated the code and it was the same--which is by design.

[the remark this answers is not in the document]
Comment (cleemullins) [reply packages/LUISGen/src/Utils.cs]: [Test]
public void IsListNullTest()
{
  Utils.Islist(null, null); // should not throw
}

**Pull request merged**